### PR TITLE
Feature/convention http server duration

### LINKED ISSUE #2298: `http.server.duration` metric description is wrong
The metric description for `http.server.duration` should be `Measures the duration of inbound HTTP requests` while we're using `measures the duration of the inbound HTTP request`.

This could cause an error like this (from my environment)

```
2024-02-21 14:00:05 2024-02-21T06:00:05.268Z    error   prometheusexporter@v0.93.0/log.go:23    error gathering metrics: collected metric http_server_duration_milliseconds label:{name:"deployment_environment" value:"local"} label:{name:"http_flavor" value:"1.0"} label:{name:"http_host" value:"[127.0.0.1:10000](http://127.0.0.1:10000/)"} label:{name:"http_method" value:"POST"} label:{name:"http_scheme" value:"http"} label:{name:"http_server_name" value:"[0.0.0.0](http://0.0.0.0/)"} label:{name:"http_status_code" value:"200"} label:{name:"job" value:"featbit-das"} label:{name:"label1" value:"featbit"} label:{name:"net_host_port" value:"10000"} histogram:{sample_count:3 sample_sum:44 bucket:{cumulative_count:0 upper_bound:0} bucket:{cumulative_count:0 upper_bound:5} bucket:{cumulative_count:0 upper_bound:10} bucket:{cumulative_count:3 upper_bound:25} bucket:{cumulative_count:3 upper_bound:50} bucket:{cumulative_count:3 upper_bound:75} bucket:{cumulative_count:3 upper_bound:100} bucket:{cumulative_count:3 upper_bound:250} bucket:{cumulative_count:3 upper_bound:500} bucket:{cumulative_count:3 upper_bound:750} bucket:{cumulative_count:3 upper_bound:1000} bucket:{cumulative_count:3 upper_bound:2500} bucket:{cumulative_count:3 upper_bound:5000} bucket:{cumulative_count:3 upper_bound:7500} bucket:{cumulative_count:3 upper_bound:10000}} has help "measures the duration of the inbound HTTP request" but should have "Measures the duration of inbound HTTP requests."
2024-02-21 14:00:05     {"kind": "exporter", "data_type": "metrics", "name": "prometheus"}
2024-02-21 14:00:05 [github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusexporter.(*promLogger).Println](http://github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusexporter.(*promLogger).Println)
2024-02-21 14:00:05     [github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusexporter@v0.93.0/log.go:23](http://github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusexporter@v0.93.0/log.go:23)
2024-02-21 14:00:05 [github.com/prometheus/client_golang/prometheus/promhttp.HandlerForTransactional.func1](http://github.com/prometheus/client_golang/prometheus/promhttp.HandlerForTransactional.func1)
2024-02-21 14:00:05     [github.com/prometheus/client_golang@v1.18.0/prometheus/promhttp/http.go:144](http://github.com/prometheus/client_golang@v1.18.0/prometheus/promhttp/http.go:144)
2024-02-21 14:00:05 net/http.HandlerFunc.ServeHTTP
2024-02-21 14:00:05     net/http/server.go:2136
2024-02-21 14:00:05 net/http.(*ServeMux).ServeHTTP
2024-02-21 14:00:05     net/http/server.go:2514
2024-02-21 14:00:05 [go.opentelemetry.io/collector/config/confighttp.(*decompressor).ServeHTTP](http://go.opentelemetry.io/collector/config/confighttp.(*decompressor).ServeHTTP)
2024-02-21 14:00:05     [go.opentelemetry.io/collector/config/confighttp@v0.93.0/compression.go:147](http://go.opentelemetry.io/collector/config/confighttp@v0.93.0/compression.go:147)
2024-02-21 14:00:05 [go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp.(*middleware).serveHTTP](http://go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp.(*middleware).serveHTTP)
2024-02-21 14:00:05     [go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.47.0/handler.go:225](http://go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.47.0/handler.go:225)
2024-02-21 14:00:05 [go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp.NewMiddleware.func1.1](http://go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp.NewMiddleware.func1.1)
2024-02-21 14:00:05     [go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.47.0/handler.go:83](http://go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.47.0/handler.go:83)
2024-02-21 14:00:05 net/http.HandlerFunc.ServeHTTP
2024-02-21 14:00:05     net/http/server.go:2136
2024-02-21 14:00:05 [go.opentelemetry.io/collector/config/confighttp.(*clientInfoHandler).ServeHTTP](http://go.opentelemetry.io/collector/config/confighttp.(*clientInfoHandler).ServeHTTP)
2024-02-21 14:00:05     [go.opentelemetry.io/collector/config/confighttp@v0.93.0/clientinfohandler.go:28](http://go.opentelemetry.io/collector/config/confighttp@v0.93.0/clientinfohandler.go:28)
2024-02-21 14:00:05 net/http.serverHandler.ServeHTTP
2024-02-21 14:00:05     net/http/server.go:2938
2024-02-21 14:00:05 net/http.(*conn).serve
2024-02-21 14:00:05     net/http/server.go:2009
```

- https://github.com/search?q=repo%3Aopen-telemetry%2Fopentelemetry-python-contrib+HTTP_SERVER_DURATION&type=code
- https://github.com/open-telemetry/opentelemetry-demo/pull/738

### PR DESCRIPTION
**Pull request by Rodrigo-Novas**
# Description

The metric description for http.server.duration should be Measures the duration of inbound HTTP requests while we're using measures the duration of the inbound HTTP request.

The same description is placed for all http.server.duration

Fixes #2298 

Since version 1.2.2 of Otel the description of http.server.duration changes to **"Duration of HTTP server requests."** for that reason instead of setting "Measures the duration of inbound HTTP requests." As the [issue](https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2298) asks I set the new description **"Duration of HTTP server requests."**

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- Run unit tests with tox

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Comment by lzchen** [comment]
Specs reference: https://github.com/open-telemetry/semantic-conventions/blob/v1.21.0/docs/http/http-metrics.md#metric-httpserverduration

**Review by lzchen — APPROVED** [review]
Changes LGTM. Please fix the conflicts and we can get this merged!

**Comment by shalevr** [comment]
> Specs reference: https://github.com/open-telemetry/semantic-conventions/blob/v1.21.0/docs/http/http-metrics.md#metric-httpserverduration

Shouldn't the description be "Duration of HTTP server requests."
according to the lastest semantic-convention(1.24.0)?

https://github.com/open-telemetry/semantic-conventions/blob/v1.24.0/docs/http/http-metrics.md#metric-httpserverrequestduration

**Review by ocelotl — APPROVED** [review]

**Comment by Rodrigo-Novas** [comment]
> Duration of HTTP server requests.

@shalevr Yes it is true, from otel>=1.22.0 the description changed, i`m going to change it.

**Comment by ocelotl** [comment]
@Rodrigo-Novas please let us know when this PR is ready for review again, marking it as draft now to prevent accidental merging.

**Comment by Rodrigo-Novas** [comment]
Thanks for the comments! 
It's ready to be reviewed again @lzchen @ocelotl @shalevr

File: instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py [hunk 1]
```diff
@@ -612,7 +612,7 @@ def instrument_app(
             duration_histogram = meter.create_histogram(
                 name=MetricInstruments.HTTP_SERVER_DURATION,
                 unit="ms",
-                description="measures the duration of the inbound HTTP request",
+                description="Measures the duration of inbound HTTP requests",
```
Comment (shalevr) [inline instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py:615]: This should be "Duration of HTTP client requests." right?

Comment (Rodrigo-Novas) [reply instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py:615]: Yes good catch!

**Pull request merged**